core/src/main/scala/org/apache/spark/partial/BoundedDouble.scala

```diff
@@ -21,5 +21,20 @@ package org.apache.spark.partial
  * A Double value with error bars and associated confidence.
  */
 class BoundedDouble(val mean: Double, val confidence: Double, val low: Double, val high: Double) {
-  override def toString(): String = "[%.3f, %.3f]".format(low, high)
+  override def toString(): String = "BoundedDouble(%.3f, %.3f, %.3f, %.3f)".format(mean, confidence, low, high)
```

Comment:
case classes bring extra concerns with binary compatibility (due to pattern matching). I'd minimize its use in public APIs.